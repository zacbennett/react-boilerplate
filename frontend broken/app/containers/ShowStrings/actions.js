/*
 *
 * ShowStrings actions
 *
 */

import {
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from './constants';


export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}
export function loadStringsSuccess(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}
export function loadStringsError(error) {
  return {
    type: LOAD_STRINGS_ERROR,
    error,
  };
}
