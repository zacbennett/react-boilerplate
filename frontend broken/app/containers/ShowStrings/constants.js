/*
 *
 * ShowStrings constants
 *
 */

export const DEFAULT_ACTION = 'app/ShowStrings/DEFAULT_ACTION';
export const LOAD_STRINGS = 'app/ShowStrings/STRINGS_LOADED';
export const LOAD_STRINGS_SUCCESS = 'app/ShowStrings/LOAD_STRINGS_SUCCESS';
export const LOAD_STRINGS_ERROR = 'app/ShowStrings/LOAD_STRINGS_ERROR';
