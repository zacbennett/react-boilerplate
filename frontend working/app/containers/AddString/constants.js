/*
 * HomeConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

// // export const CHANGE_USERNAME = 'boilerplate/Home/CHANGE_USERNAME';
export const CHANGE_STRING = 'boilerplate/Home/CHANGE_STRING';
export const UPLOAD_STRING = 'boilerplate/Home/UPLOAD_STRING';
export const UPLOAD_STRING_SUCCESS = 'boilerplate/Home/UPLOAD_STRING_SUCCESS';
export const UPLOAD_STRING_ERROR = 'boilerplate/Home/UPLOAD_STRING_ERROR';
export const CLEAR_SUCCESS_ERROR = 'boilerplate/Home/CLEAR_SUCCESS_ERROR';
