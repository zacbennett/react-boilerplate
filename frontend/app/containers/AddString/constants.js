/*
 *
 * AddString constants
 *
 */

export const DEFAULT_ACTION = 'app/AddString/DEFAULT_ACTION';
export const ADD_STRING = 'app/AddString/ADD_STRING';
