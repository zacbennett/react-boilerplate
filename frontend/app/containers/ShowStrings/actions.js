/*
 *
 * ShowStrings actions
 *
 */

import {
  DEFAULT_ACTION,
  LOAD_STRINGS,
  LOAD_STRINGS_SUCCESS,
  LOAD_STRINGS_ERROR,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function loadStrings() {
  return {
    type: LOAD_STRINGS,
  };
}
export function loadStringsSuccess(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}
export function loadStringsError() {
  return {
    type: LOAD_STRINGS_ERROR,
  };
}
