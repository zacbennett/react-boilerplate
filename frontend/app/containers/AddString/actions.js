/*
 *
 * AddString actions
 *
 */

import { DEFAULT_ACTION, ADD_STRING } from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}
export function addString(data) {
  return {
    type: ADD_STRING,
    data,
  };
}
