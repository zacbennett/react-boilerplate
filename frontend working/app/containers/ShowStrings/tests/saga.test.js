/**
 * Tests for ShowStrings sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_STRINGS } from '../constants';
import { loadStringsSuccess, loadStringsError } from '../actions';

import stringData, { getStrings } from '../saga';

// const username = 'mxstbr';

// /* eslint-disable redux-saga/yield-effects */
describe('showStrings Saga', () => {
  let loadStringsGenerator;

  beforeEach(() => {
    loadStringsGenerator = getStrings();

    const selectDescriptor = loadStringsGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the reposLoaded action if it requests the data successfully', () => {
    const response = [{ id: 1, data: 'hi' }];
    const putDescriptor = loadStringsGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(loadStringsSuccess(response)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = loadStringsGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loadStringsError(response)));
  });

  describe('makeAPICallForStrings Saga', () => {
    const makeAPICallSaga = stringData();

    it('should start task to watch for LOAD_STRING action', () => {
      const takeLatestDescriptor = makeAPICallSaga.next().value;
      expect(takeLatestDescriptor).toEqual(
        takeLatest(LOAD_STRINGS, getStrings),
      );
    });
  });
});
