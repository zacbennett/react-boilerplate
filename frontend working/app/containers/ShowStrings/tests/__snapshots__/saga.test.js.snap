// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRepos Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`getRepos Saga should dispatch the reposLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`showStrings Saga makeAPICallForStrings Saga should start task to watch for LOAD_STRING action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`showStrings Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`showStrings Saga should dispatch the reposLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "http://localhost:3001/strings",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
